tests/ui/async-await/async-closures/dyn.stderr

```diff
@@ -0,0 +1,83 @@
+error[E0802]: the trait `AsyncFn` is not yet dyn-compatible
+  --> $DIR/dyn.rs:10:23
+   |
+LL | fn takes_async_fn(_: &dyn AsyncFn()) {}
+   |                       ^^^^^^^^^^^^^ `AsyncFn` is not yet dyn compatible
+   |
+help: consider using an opaque type instead
+   |
+LL | fn takes_async_fn(_: &impl AsyncFn()) {}
+   |                       ~~~~
+
+error[E0802]: the trait `AsyncFnMut` is not yet dyn-compatible
+  --> $DIR/dyn.rs:13:31
+   |
+LL | fn takes_async_fn_mut(_: &mut dyn AsyncFnMut()) {}
+   |                               ^^^^^^^^^^^^^^^^ `AsyncFnMut` is not yet dyn compatible
+   |
+help: consider using an opaque type instead
+   |
+LL | fn takes_async_fn_mut(_: &mut impl AsyncFnMut()) {}
+   |                               ~~~~
+
+error[E0802]: the trait `AsyncFnOnce` is not yet dyn-compatible
+  --> $DIR/dyn.rs:16:31
+   |
+LL | fn takes_async_fn_once(_: Box<dyn AsyncFnOnce()>) {}
+   |                               ^^^^^^^^^^^^^^^^^ `AsyncFnOnce` is not yet dyn compatible
+   |
+help: consider using an opaque type instead
+   |
+LL | fn takes_async_fn_once(_: Box<impl AsyncFnOnce()>) {}
+   |                               ~~~~
+
+error[E0802]: the trait `AsyncFn` is not yet dyn-compatible
+  --> $DIR/dyn.rs:22:36
+   |
+LL | fn takes_async_fn_implicit_dyn(_: &AsyncFn()) {}
+   |                                    ^^^^^^^^^ `AsyncFn` is not yet dyn compatible
+   |
+help: consider using an opaque type instead
+   |
+LL | fn takes_async_fn_implicit_dyn(_: &impl AsyncFn()) {}
+   |                                    ++++
+
+error[E0802]: the trait `AsyncFnMut` is not yet dyn-compatible
+  --> $DIR/dyn.rs:26:44
+   |
+LL | fn takes_async_fn_mut_implicit_dyn(_: &mut AsyncFnMut()) {}
+   |                                            ^^^^^^^^^^^^ `AsyncFnMut` is not yet dyn compatible
+   |
+help: consider using an opaque type instead
+   |
+LL | fn takes_async_fn_mut_implicit_dyn(_: &mut impl AsyncFnMut()) {}
+   |                                            ++++
+
+error[E0802]: the trait `AsyncFnOnce` is not yet dyn-compatible
+  --> $DIR/dyn.rs:30:44
+   |
+LL | fn takes_async_fn_once_implicit_dyn(_: Box<AsyncFnOnce()>) {}
+   |                                            ^^^^^^^^^^^^^ `AsyncFnOnce` is not yet dyn compatible
+   |
+help: consider using an opaque type instead
+   |
+LL | fn takes_async_fn_once_implicit_dyn(_: Box<impl AsyncFnOnce()>) {}
+   |                                            ++++
+
+error[E0038]: the trait `SubAsyncFn` cannot be made into an object
```

Comment:
We already have an error code for object safety, why add another one? Also, I am a bit unsatisfied with the inconsistency in the error messaging here.

Users don't really need an in-depth explanation, and it's something that I expect for us to fix in the medium-term future with my introduction of general AFIDT support.

Reply:
I didn't use the language "cannot be made into an object" in the new error message because my understanding was that "object-safety" is the deprecated term, and "dyn-compatibility" is the new one. I also didn't change both to use "dyn-compatibility" because that would have affected significantly more code, though I can send a separate PR for that if it is desired.

Reply:
I think just making this say the same thing as what E0038 says by default is fine, and if you'd like to reword the general error message in a separate PR it should be done separately.

Reply:
IMO the net effect of this PR should be to deduplicate the many different dyn compatibility violations, and to add a note explaining that AsyncFn* is not. 

Adding a totally different error message and a new error code seem unnecessary on top of that I feel?